lib/ansible/modules/cloud/amazon/ecs_taskdefinition.py

```diff
@@ -435,7 +438,8 @@ def _task_definition_matches(requested_volumes, requested_containers, existing_t
             for td in existing_definitions_in_family:
                 requested_volumes = module.params.get('volumes', []) or []
                 requested_containers = module.params.get('containers', []) or []
-                existing = _task_definition_matches(requested_volumes, requested_containers, td)
+                requested_task_role_arn = module.params.get('task_role_arn', "") or ""
```

Comment:
I think `requested_task_role_arn = module.params['task_role_arn']` will work here.

The default in module.params is an empty string so you don't really need `.get('task_role_arn', "")` (the .get is also unused on the two lines above this, since `{'mykey': None}.get('mykey', [])` does not do what `{}.get('mykey', [])` does). And since a False string is an empty one, you can also remove the `or ""`.